api/src/main/java/org/pmiops/workbench/cohortbuilder/util/QueryParameterValues.java

```diff
@@ -4,58 +4,96 @@
 import com.google.cloud.bigquery.QueryParameterValue;
 import com.google.cloud.bigquery.StandardSQLTypeName;
 import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
 import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
+import java.util.Locale;
 import java.util.Map;
 import java.util.Optional;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
+import javax.annotation.Nullable;
 import org.hibernate.engine.jdbc.internal.BasicFormatterImpl;
+import org.jetbrains.annotations.NotNull;
 import org.pmiops.workbench.utils.Matchers;
 
 public final class QueryParameterValues {
   private static final int MICROSECONDS_IN_MILLISECOND = 1000;
 
   // For creating a Timestamp QueryParameterValue, use this formatter.
   // example error when using the RowToInsert version (below): "Invalid format:
-  // "1989-02-17 00:00:00.000000" is too short".
+  // "1989-02-17 00:00:00.000000" is too short". See https://stackoverflow.com/a/55155067/611672
+  // for a nice walkthrough of the machinations involved.
   public static final DateTimeFormatter QPV_TIMESTAMP_FORMATTER =
-      DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSSSSSZZ").withZone(ZoneOffset.UTC);
+      DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSSSSSxxx");
+
   // For inserting a Timestamp in a RowToInsert map for an InsertAllRequest, use this format
   public static final DateTimeFormatter ROW_TO_INSERT_TIMESTAMP_FORMATTER =
-      DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSSSSS").withZone(ZoneOffset.UTC);
+      DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSSSSS")
+          .withZone(ZoneOffset.UTC)
+          .withLocale(Locale.US);
 
   /** Generate a unique parameter name and add it to the parameter map provided. */
+  @NotNull
```

Comment:
Since this class has two or three different nullity contracts, I tried to be explicit with `@NotNull` and `@Nulllable`. I'm not necessarily advocating we do that everywhere.